openpype/hosts/maya/plugins/publish/extract_workfile_xgen.py

```diff
@@ -0,0 +1,250 @@
+
+         Returns:
+            tuple or None: Start frame, end frame tuple if any renderlayers
+                found. Otherwise None is returned.
+
+         """
+
+        def _is_active_renderlayer(i):
+            """Return whether instance is active renderlayer"""
+            if not i.data.get("publish", True):
+                return False
+
+            is_renderlayer = (
+                "renderlayer" in i.data.get("families", []) or
+                i.data["family"] == "renderlayer"
+            )
+            return is_renderlayer
+
+        start_frame = None
+        end_frame = None
+        for instance in context:
+            if not _is_active_renderlayer(instance):
+                # Only consider renderlyare instances
+                continue
+
+            render_start_frame = instance.data["frameStart"]
+            render_end_frame = instance.data["frameStart"]
+
+            if start_frame is None:
+                start_frame = render_start_frame
+            else:
+                start_frame = min(start_frame, render_start_frame)
+
+            if end_frame is None:
+                end_frame = render_end_frame
+            else:
+                end_frame = max(end_frame, render_end_frame)
+
+        if start_frame is None or end_frame is None:
+            return
+
+        return start_frame, end_frame
+
+    def process(self, instance):
+        transfers = []
+
+        # Validate there is any palettes in the scene.
+        if not cmds.ls(type="xgmPalette"):
+            self.log.debug(
+                "No collections found in the scene. Skipping Xgen extraction."
+            )
+            return
+        else:
+            import xgenm
+
+        # Validate to extract only when we are publishing a renderlayer as
+        # well.
+        render_range = self.get_render_max_frame_range(instance.context)
+        if not render_range:
+            self.log.debug(
+                "No publishable renderlayers found in context. Skipping Xgen"
+                " extraction."
+            )
+            return
+
+        start_frame, end_frame = render_range
+
+        # We decrement start frame and increment end frame so motion blur will
+        # render correctly.
+        start_frame -= 1
+        end_frame += 1
+
+        # Extract patches alembic.
+        basename, _ = os.path.splitext(instance.context.data["currentFile"])
+        dirname = os.path.dirname(instance.context.data["currentFile"])
+        kwargs = {"attrPrefix": ["xgen"], "stripNamespaces": True}
+        alembic_files = []
+        for palette in cmds.ls(type="xgmPalette"):
+            patch_names = []
+            for description in xgenm.descriptions(palette):
+                for name in xgenm.boundGeometry(palette, description):
+                    patch_names.append(name)
+
+            alembic_file = os.path.join(
+                dirname,
+                "{}__{}.abc".format(basename, palette.replace(":", "__ns__"))
```

Comment:
Its the `basename` of the current workfile, which needs to be included in the alembic file name.

Reply:
So `instance.context.data["currentFile"])` is the full path to the current file as far as I know: `C:/path/to/maya/scene.ma`
Doing `basename, _ = os.path.splitext(current_file)` on it makes `basename` `C:/path/to/maya/scene`.

As such we appear to be joining that absolute path onto the dirname here?

I think looking at the code you intended to originally call `os.path.basename` as well as `os.path.splitext` on the current file?

Reply:
Ahh I see, yeah you are right.

Reply:
So it is intentional that you're joining the full path onto that dirname generating a path like:
```
C:/path/to/dirname/C:/path/to/maya/currentfile_{palette}.abc
```

Because:
```python
            alembic_file = os.path.join(
                dirname,
                "{}__{}.abc".format(
                    path_no_ext, palette.replace(":", "__ns__")
                )
            )
```
Because:
```python
dirname = "C:/path/to/current"
path_no_ext = "C:/path/to/current/scene"
```

Reply:
yeah, good point. Simplified now.